pulpcore/app/migrations/0053_add_public_key.py

```diff
@@ -0,0 +1,105 @@
+# Generated by Django 2.2.17 on 2020-12-05 11:11
+
+import hashlib
+import json
+import warnings
+import subprocess
+import tempfile
+
+import gnupg
+
+from django.db import migrations, models
+
+
+def sign(service, filename):
+    completed_process = subprocess.run(
+        [service.script, filename], env={}, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+    )
+
+    if completed_process.returncode != 0:
+        raise RuntimeError()
+
+    try:
+        return_value = json.loads(completed_process.stdout)
+    except json.JSONDecodeError:
+        raise RuntimeError()
+
+    return return_value
+
+
+def print_warning(service):
+    warnings.warn(
+        "The public key migration for '{}' failed. Consider creating a new signing "
+        "service instead and try to run the migration again. The resource will not "
+        "be valid.".format(service.name),
```

Comment:
You cannot "run the migration again".
How about:
```suggestion
        "The public key migration for '{}' failed.  The resource will not "
        "be valid. Consider creating a new signing service instead."
        "".format(service.name),
```